src/ipv4.h

```diff
@@ -18,6 +18,8 @@
 #ifndef _OPENFORTIVPN_IPV4_H
 #define _OPENFORTIVPN_IPV4_H
 
+#include <sys/types.h>
+#include <sys/socket.h>
```

Comment:
Out of curiosity, what are `sys/types.h` and `sys/socket.h` needed for? Perhaps they should be added to source files instead of this header file.
Is this related to `rtentry`?

Reply:
yes, this was following the synopsis of https://www.freebsd.org/cgi/man.cgi?query=rtentry&sektion=9 but in the meantime I have noticed that OpenBSD and FreeBSD, compared to linux, both use a different implementation for `rtentry`

Reply:
Like OS X perhaps?

If so perhaps we can re-use the OS X impelmentation with BSDs? Then the `#ifdef __APPLE__` could be changed to something like `#ifndef HAVE_RTRENTRY` or something like this. Is there something like this in _autoconf_?

Otherwise, yes, the additional headers should be guarded (if possible) by something like `#ifdef __BSD__`.

Reply:
The [`configure.in` from _tcpdump_](https://opensource.apple.com/source/tcpdump/tcpdump-79.60.1/tcpdump/configure.in.auto.html) might be a good source of inspiration.

Reply:
I think the situation for OSX is different. We provide the definition of `rtentry` for OSX. For BSDs there is already a struct with that name and slightly different fields. With C we can't simply extend or overwrite that structure, but I still have the hope that I can define a few macros accordingly. Once the code compiles I can put around some `#ifdef` constructs, hopefully smart ones that use the outcome of configure tests